### ISSUE TITLE
Hello, I followed your experimental steps to reproduce the process, but the loss remains the same. Do you have a solution? Or any information about the parameter settings?

### ISSUE THREAD

**Issue by FlAmmmmING** [matt-seb-ho/gfn_ntp#5]
![image](https://github.com/user-attachments/assets/c99e7884-97e5-4e16-b9df-73cc09d0108c)

**Comment by FlAmmmmING** [comment]
Another issue is that when I set use_lora to true in reprover_full_finetune, the program throws an error: omegaconf.errors.InterpolationKeyError: Interpolation key 'architecture' not found.


`
hf_id: kaiyuy/leandojo-lean4-tacgen-byt5-small
architecture: ${shared_values.architecture}
seq2seq: ${shared_values.seq2seq}
use_lora: true # I modified here
use_4bit: ${shared_values.use_4bit}

# specify a huggingface model id to initialize the policy adapter from
initial_policy_adapter: null

lora_config:
  _target_: peft.LoraConfig

  # adjusted settings for OOM issues
  target_modules: all-linear
  r: 32
  lora_alpha: 32
  lora_dropout: 0.05
  bias: none
  task_type: ${arch2task_type[${.architecture}]}
  use_rslora: true

bnb:
  _target_: transformers.BitsAndBytesConfig
  load_in_4bit: true
  bnb_4bit_use_double_quant: true
  bnb_4bit_quant_type: nf4
  bnb_4bit_compute_dtype: bfloat16

# used for variable interpolation
arch2task_type:
  seq2seq: SEQ_2_SEQ_LM
  causal: CAUSAL_LM

`

**Comment by matt-seb-ho** [comment]
Hi, thanks for reaching out! Regarding the missing architecture value, the top level `config.yaml` should have a `shared_values` section that defines the architecture type variable. The example one looks like this:
```yaml
# values used across multiple config modules
shared_values:
  repeat_theorem_n_times: 1
  architecture: seq2seq # {seq2seq, causal}
  seq2seq: ${arch2seq2seq[${shared_values.architecture}]}
  use_4bit: false
```

Regarding the training loss, we also had some trouble stabilizing this loss. This is still a work in progress so we intend to keep scaling and trying new configurations, but the current ones described in the paper do seem to converge. A key intervention we used was initializing the replay buffer to contain the ground truth as a work-around for the high search budget that's otherwise needed for the initial policy to find high-reward solutions.

**Comment by FlAmmmmING** [comment]
Thank you for your reply! I will use your method to run the experiment again! However, I also look forward to you releasing new parameter configurations!
If I have any other questions, I will come back to ask you!

**Comment by FlAmmmmING** [comment]
![image](https://github.com/user-attachments/assets/f40428a1-0a50-4a0b-8a71-bc2418ad0698)
Hello, this issue still persists. Could it be because I haven't properly tuned the parameters?

**Comment by FlAmmmmING** [comment]
Could you please share the parameter settings you used in the paper?https://arxiv.org/abs/2410.13224

**Comment by matt-seb-ho** [comment]
>Paper Parameter Settings

The current published configs on the repo are the same settings used in the paper. This represents the `GFlowNet` setting in Figures 2 and 3. For "BR" binary reward setting and "OO" offline only, these are trivial adjustments to the current config (binary reward is changing `reward.model.setup` to `None`, and offline only involve setting `ntp.ntp_config.user_buffer_prob` to 0).

>`Interpolation key 'architecture' not found` error persists

Can you please try replacing `task_type: ${arch2task_type[${.architecture}]}` in `configs/model/reprover_full_finetune.yaml` with `task_type: ${arch2task_type[${..architecture}]}` or `task_type: ${arch2task_type[${shared_values.architecture}]}`?

**Comment by FlAmmmmING** [comment]
it throws another error like this:
<img width="1000" alt="image" src="https://github.com/user-attachments/assets/29e1ad79-7241-4a53-ae1c-1865e095ff4e" />

**Comment by matt-seb-ho** [comment]
Please try  `task_type: ${..arch2task_type[${..architecture}]}` or `task_type: ${model.arch2task_type[${..architecture}]}`.

**Comment by FlAmmmmING** [comment]
<img width="1339" alt="image" src="https://github.com/user-attachments/assets/f044fe67-5e1f-46a9-8c30-837c8786344d" />

<img width="1245" alt="image" src="https://github.com/user-attachments/assets/b3b08f99-edde-4f60-850c-462417932e05" />

Hello! I’ve been trying to reproduce your experiment over the past few days, but I found that the program suddenly throws an error and stops training halfway through. Do you know why this might be happening?

**Comment by FlAmmmmING** [comment]
This is the whole error.
<img width="1354" alt="image" src="https://github.com/user-attachments/assets/e5bbd252-753e-4cdc-b3f5-87e243de8fbd" />
<img width="1411" alt="image" src="https://github.com/user-attachments/assets/0c6013e7-de55-4714-97ff-1f71e94e300c" />
<img width="1255" alt="image" src="https://github.com/user-attachments/assets/333de11a-01ad-492c-a703-100f1ae4f79e" />

And when I want to run test_eval or test_gfn, it occurs:
<img width="1283" alt="image" src="https://github.com/user-attachments/assets/0a2c1495-37f1-47e8-9ca3-02564beb81b1" />

Could you pleas help me?

**Comment by FlAmmmmING** [comment]
Hello! I’ve been trying to reproduce your experiment over the past few days, but I found that the program suddenly throws an error and stops training halfway through. Do you know why this might be happening?
This is the whole error.
```cmd
best first proof search: 100%|██████████████████| 20/20 [06:02<00:00, 18.12s/it]earch: 100%|████████
_ÿ█████████| 20/20 [06:02<00:00, 22.91s/it]                                                              Error executin
g job with overrides: []                   | 0/20 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/lean_dojo/interaction/dojo.py", line 418, in _read_next_
line
    index = self.proc.expect(["\n", f"{_REPL_PROMPT}.*?\n"])
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pexpect/spawnbase.py", line 354, in expect
    return self.expect_list(compiled_pattern_list,
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pexpect/spawnbase.py", line 383, in expect_list
    return exp.expect_loop(timeout)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pexpect/expect.py", line 181, in expect_loop
    return self.timeout(e)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pexpect/expect.py", line 144, in timeout
    raise exc
pexpect.exceptions.TIMEOUT: Timeout exceeded.
<pexpect.pty_spawn.spawn object at 0x7e199a0bbeb0>
command: /home/jwj/.elan/bin/lake
args: [b'/home/jwj/.elan/bin/lake', b'env', b'lean', b'--threads=1', b'--memory=32768', b'Mathlib/Data/Finset/Cardyblztu49.lean
']
buffer (last 100 chars): ''
before (last 100 chars): ''
after: <class 'pexpect.exceptions.TIMEOUT'>
match: None
match_index: None
exitstatus: None
flag_eof: False
pid: 2234269
child_fd: 56
closed: False
timeout: 30
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 1
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('\n')
    1: re.compile('REPL>.*?\n') 
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jwj/gfn_ntp/proof_flow/src/gfn_tuning/ntp.py", line 356, in parallel_forward_step
    next_state = lean_env.run_tac(prev_state, tactic.rstrip())
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/lean_dojo/interaction/dojo.py", line 332, in run_tac
    res = self._submit_request(req)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/lean_dojo/interaction/dojo.py", line 382, in _submit_req
uest
    res, msg = self._read_next_line()
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/lean_dojo/interaction/dojo.py", line 433, in _read_next_
line
    raise DojoTacticTimeoutError()
lean_dojo.interaction.dojo.DojoTacticTimeoutError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jwj/gfn_ntp/proof_flow/scripts/gfn_tuning/train.py", line 46, in train
    trainer.fit(model=task, datamodule=data)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 538, in fit
    call._call_and_handle_interrupt(
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 47, in _call_an
d_handle_interrupt
    return trainer_fn(*args, **kwargs)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 574, in _fit
_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 981, in _run
    results = self._run_stage() 
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 1025, in _run_stage
    self.fit_loop.run()
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/loops/fit_loop.py", line 205, in run
    self.advance()
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/loops/fit_loop.py", line 363, in advance
    self.epoch_loop.run(self._data_fetcher)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/loops/training_epoch_loop.py", line 141, in run
    self.on_advance_end(data_fetcher)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/loops/training_epoch_loop.py", line 295, in on_advance_end
    self.val_loop.run()
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/loops/utilities.py", line 178, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py", line 135, in run
self._evaluation_step(batch, batch_idx, dataloader_idx, dataloader_iter)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/loops/evaluation_loop.py", line 396, in _evaluation_step
    output = call._call_strategy_hook(trainer, hook_name, *step_args)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 319, in _call_strategy_hook
    output = fn(*args, **kwargs)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/pytorch_lightning/strategies/strategy.py", line 411, in validation_step
    return self.lightning_module.validation_step(*args, **kwargs)
  File "/home/jwj/gfn_ntp/proof_flow/src/gfn_tuning/ntp.py", line 561, in validation_step
    log_pf, log_r, _ = self.parallel_forward(theorem)
  File "/home/jwj/gfn_ntp/proof_flow/src/gfn_tuning/ntp.py", line 210, in parallel_forward
    r = self.parallel_forward_step(
  File "/home/jwj/gfn_ntp/proof_flow/src/gfn_tuning/ntp.py", line 358, in parallel_forward_step
    self._debug_log(
  File "/home/jwj/gfn_ntp/proof_flow/src/gfn_tuning/ntp.py", line 950, in _debug_log
    logger.log(self.cfg.debug_log_level, msg, depth=1)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/loguru/_logger.py", line 2103, in log
    __self._log(__level, False, __self._options, __message, args, kwargs)
  File "/home/jwj/miniconda3/envs/gfn_ntp/lib/python3.10/site-packages/loguru/_logger.py", line 2055, in _log
    log_record["message"] = message.format(*args, **kwargs)
KeyError: ' contextual '
```

And when I want to run test_eval or test_gfn, it occurs:

```cmd
(gfn_ntp) dell-PowerEdge-R7525% python -m proof_flow.scripts.gfn_tuning.test_gfn
[2024-12-27 22:51:34,442] [INFO] [real_accelerator.py:222:get_accelerator] Setting ds_accelerator to cuda (auto detect)
Cannot find primary config 'train'. Check that it's in your config search path.

Config search path:
        provider=hydra, path=pkg://hydra.conf
        provider=main, path=file:///home/jwj/gfn_ntp/configs
        provider=schema, path=structured://
```
could you help me?
